### ISSUE TITLE
Error on API calls with null response in RequestException

### ISSUE THREAD

**Issue by valigara** [jlevers/selling-partner-api#235]
https://github.com/jlevers/selling-partner-api/blob/f6ab5328351cf8b36b692b697596a34b7fbc607b/lib/Api/CatalogItemsV0Api.php#L157
produces error in case `getResponse()` returns `null` and it can happen (and happens)

```
Call to a member function getBody() on null
#0 /www/jewelry/include/composer/vendor/jlevers/selling-partner-api/lib/Api/CatalogItemsV0Api.php(928): SellingPartnerApi\Api\CatalogItemsV0Api->listCatalogItemsWithHttpInfo('ATVPDKIKX0DER', NULL, NULL, 'DDRAGRSC040-20', NULL, NULL, NULL, NULL)

```

 according to `\GuzzleHttp\Exception\RequestException`

response can be `null` please see

https://github.com/guzzle/guzzle/blob/82ca75f0b1f130f018febdda29af13086da5dbac/src/Exception/RequestException.php#L35

so my suggestion to check maybe api generation tool has already fix for such issue & regenerate or maybe to create issue ticket for that tool & after they will fix it regenerate api.

**Comment by sunil11475-dev** [comment]
Hi

If you looking for header and full response from RequestException catch then just 

```
} catch (Exception $e) {
	echo 'Exception when calling OrdersV0Api->getOrders: ', $e->getMessage(), PHP_EOL;
       $body = (string) $e;
}
print_r($body); //or each
```

**Comment by jlevers** [comment]
Thanks for reporting -- I maintain all my own API generation templates for this, so I should be able to fix this there.

**Comment by jlevers** [comment]
@valigara do you have the latest version of the library installed (`v4.3.8`)? I can't reproduce your issue.

**Comment by valigara** [comment]
@jlevers no, currently we on 4.3.3 
but ive checked with 4.3.8 it also has same issue
here my test script that allows to reproduce issue:
```php
class TestClient extends \GuzzleHttp\Client
{

    public function send(\Psr\Http\Message\RequestInterface $request, array $options = []): \Psr\Http\Message\ResponseInterface
    {
        if (stripos($request->getUri()->getPath(), 'marketplaceParticipations') !== false) {
            throw new \GuzzleHttp\Exception\RequestException('Test RequestException with empty Response', $request);
        }
        return parent::send($request, $options);
    }

}

$client = new TestClient();
$api = new \SellingPartnerApi\Api\SellersApi($config, $client);
$result = $api->getMarketplaceParticipations();
```
Output:
```
Fatal error:  Uncaught Error: Call to a member function getBody() on null in /var/www/ldev/test/vendor/jlevers/selling-partner-api/lib/Api/SellersApi.php:153
Stack trace:
#0 /var/www/ldev/test/vendor/jlevers/selling-partner-api/lib/Api/SellersApi.php(124): SellingPartnerApi\Api\SellersApi->getMarketplaceParticipationsWithHttpInfo()
#1 /var/www/ldev/test/index.php(81): SellingPartnerApi\Api\SellersApi->getMarketplaceParticipations()
#2 {main}
  thrown in /var/www/ldev/test/vendor/jlevers/selling-partner-api/lib/Api/SellersApi.php on line 153

```

**Comment by jlevers** [comment]
This should be fixed in `v4.3.10`. Let me know if you have any more issues with this.

**Comment by valigara** [comment]
@jlevers we are highly appreciated for this fix & your time.

But [null safe operator](https://wiki.php.net/rfc/nullsafe_operator) `?->` only available in php version 8.0 and newer,
library itself decelerates php 7.3 support for some peoples that are stuck with older versions like php 7.4 it will be `Parse error`.

For example we can't migrate to `v4.3.10` :
![image](https://user-images.githubusercontent.com/40270518/163716370-213939ce-649b-4d7e-a84b-813047869dcc.png)

we are on 7.4 yet, also it may cause huge problems for other peoples with older php versions.
